device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsDeviceAuthenticationCBS.java

```diff
@@ -176,6 +176,9 @@ protected Boolean authenticationMessageReceived(AmqpsMessage amqpsMessage, UUID
                         int propertyValue = (int) entry.getValue();
                         if (propertyValue == 200)
                         {
+                            // Codes_SRS_AMQPSDEVICEAUTHENTICATIONCBS_34_035: [If the correlationId and status code matches, this function will acknowledge the provided amqpsMessage.]
+                            amqpsMessage.acknowledge(AmqpsMessage.ACK_TYPE.COMPLETE);
```

Comment:
this is a fix for both DC and MC... DC e2e wasn't catching this previously?

Reply:
This fix is only relevant for MC/DC against edgehub, so no e2e test existed for it in our gate. It will be soon, though